concordium-consensus/src/Concordium/GlobalState/Persistent/Account.hs

```diff
@@ -36,9 +36,12 @@ import qualified Concordium.GlobalState.Basic.BlockState.Account as Transient
 import qualified Concordium.GlobalState.Basic.BlockState.AccountReleaseSchedule as Transient
 import Concordium.GlobalState.Persistent.BlobStore
 import Concordium.GlobalState.Persistent.BlockState.AccountReleaseSchedule
+import Concordium.GlobalState.Persistent.CachedRef
 import Concordium.GlobalState.Account hiding (addIncomingEncryptedAmount, addToSelfEncryptedAmount)
 
 -- | The persistent version of the encrypted amount structure per account.
+-- We use 'EagerBufferedRef's for the encrypted amounts so that when a 'PersistentAccount' is
+-- loaded, the entire encrypted amount will also be loaded.
```

Comment:
While this states the "what", the more important thing is the "why". It cannot be because of expected usage pattern, so it must be due to hashing. Please document intent.